core/include/io.h

```diff
@@ -20,6 +20,25 @@
 #define READ_ONCE(p)		__compiler_atomic_load(&(p))
 #define WRITE_ONCE(p, v)	__compiler_atomic_store(&(p), (v))
 
+/*
+ *  IO port access primitives
+ *  -------------------------
+ *
+ * Arm A-profile cores don't have special IO access instructions.
+ * OP-TEE maps IO memory as device, shareable and bufferable but not cacheable.
+ *
+ * io_readX(), io_writeX() io_clr/setbitsX() functions do not guarantee
+ * that the IO request reached the device in the code order.
+ *
+ * The barrier version of each function - io_readX_barrier(), io_writeX_barier()
+ * and io_clr/setbitsX_barrier() - guarantee that IO access are
+ * observable for system (outer shareable):
+ * - Make sure the write above is visible
+ * - Make sure the read is performed
+ */
+#define __rmb()		dsb()
```

Comment:
`core/include/io.h` is an architecture common file so we should at the very least surround this with ifdefs. Then we'd also have to do the same with an `#include <arm.h>` above so perhaps it's better to put this in `core/arch/arm/include/io_arch.h` or such.

I'd prefer a more descriptive name for these barriers, such as `io_before_write_barrier()`, `io_after_write_barrier()`, and so on.

Shouldn't this be a `dsb_oshld` and the write barrier `dsb_oshst`?